classes/Http/Controller/TalkProcessingController.php

```diff
@@ -0,0 +1,279 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Controller;
+
+use HTMLPurifier;
+use OpenCFP\Domain\CallForPapers;
+use OpenCFP\Domain\Model\Talk;
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Http\Form\TalkForm;
+use OpenCFP\Http\View\TalkHelper;
+use Swift_Mailer;
+use Swift_Message;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig_Environment;
+
+class TalkProcessingController extends BaseController
+{
+    private $authentication;
+    private $talkHelper;
+    private $callForPapers;
+    private $purifier;
+    private $mailer;
+    private $applicationEmail;
+    private $applicationTitle;
+    private $applicationEndDate;
+
+    public function __construct(
+        Authentication $authentication,
+        TalkHelper $talkHelper,
+        CallForPapers $callForPapers,
+        HTMLPurifier $purifier,
+        Swift_Mailer $mailer,
+        Twig_Environment $twig,
+        UrlGeneratorInterface $urlGenerator,
+        string $applicationEmail,
+        string $applicationTitle,
+        string $applicationEndDate
+    ) {
+        $this->authentication     = $authentication;
+        $this->talkHelper         = $talkHelper;
+        $this->callForPapers      = $callForPapers;
+        $this->purifier           = $purifier;
+        $this->mailer             = $mailer;
+        $this->applicationEmail   = $applicationEmail;
+        $this->applicationTitle   = $applicationTitle;
+        $this->applicationEndDate = $applicationEndDate;
+
+        parent::__construct($twig, $urlGenerator);
+    }
+
+    public function processCreateAction(Request $request)
+    {
+        // You can only create talks while the CfP is open
+        if (!$this->callForPapers->isOpen()) {
+            $request->getSession()->set(
+                'flash',
+                [
+                    'type'  => 'error',
+                    'short' => 'Error',
+                    'ext'   => 'You cannot create talks once the call for papers has ended', ]
+            );
+
+            return $this->redirectTo('dashboard');
+        }
+
+        $user = $this->authentication->user();
+
+        $requestData = [
```

Comment:
Most of the suggestions I make here for the *newly created* classes  are only because they are newly created classes, and as such, I think it's a good opportunity to clean up a few things. 🤓 

What do you think about inlining `$requestData` here? It's used once only.